script.js

```diff
@@ -1,12 +1,216 @@
+
+const containerEl = document.getElementById("episode-container");
+const rootElem = document.getElementById("root");
+const searchBar = document.getElementById("search-item");
+const displayCount = document.getElementById("search-result");
+const selectElement = document.getElementById("select-episode");
+const showSelector = document.getElementById("select-show");
+let showSearchBar = document.getElementById("search-show");
+showSearchBar.addEventListener("keyup", searchShows);
+
+showSelector.addEventListener("change", getEpisodes);
+
+// alphabetical order of shows
+allShows.sort((a, b) => {
+  let showOne = a.name.toLowerCase();
+  let showTwo = b.name.toLowerCase();
+
+  if (showOne < showTwo) return -1;
+  if (showOne > showTwo) return 1;
+});
+//---Getting Show page
+/* 
+  In this section you're trying to get all shows page, or home page
+  setup is called on load
+  getShows is called on search
+  */
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  hideEpisdoesRelatedParts();
+  allShows.forEach((show) => createCardForShow(show));
+  buildShowOptions(allShows);
+}
+
+function getShows(arrayOfShows) {
```

Comment:
It looks like `setup` and `getShows` are doing almost the same thing, you could probably reuse the logic.